### ISSUE TITLE
Rename "meters" and "meteringPoints"

### ISSUE THREAD

**Issue by AMEE** [AMEE/AMON#3]
Feedback is that "meters" and "meteringPoints" could be better named -- perhaps to "devices" and "virtualDevices"? Further discussion is required as to the exact names....

**Comment by revenco-hk** [comment]
Having been involved in the original discussions on the AMON format I think the specification has a conflict between "meters" and "meteringPoints". As far as I remember the idea was that the point where a measurement is taken is a Meter Point. The Meter Point may have some attributes like MPAN for fiscal electric meters. Meter Points are part of Entities (addresses/premises/groups).
In turn the Meter Point has a Meter that actually takes the measurement(s). This Meter could be a humidity sensor, electric meter, inverter, wind direction sensor etc. The reason why the Meter Point is not the same as the Meter is that the later can be changed/modified i.e. has a history whilst the Meter Point is essentially static.

**Comment by AMEE** [comment]
Hi! Yes, that's absolutely correct, and this functionality will be retained -- what is currently called a "meter" is a physical (or virtual) device that generates "measurements". However, feedback we have had is that the word "meter" in the industry is often interpreted to mean a gas meter, or an electricity meter. Things like relative humidity & temperature sensors are not commonly called "meters", although the AMON standard calls these devices "meters" as well. 

Thus, there is a desire to change the "meter" in AMON to be a "device".

As a result, if we are renaming "meter" to "device", the name "meteringPoint" is somewhat acceptable (as, as you say, the original idea was that this would represent the point where actual metering/billing is performed, using an attached "meter" or "device"). However, the scope of "meteringPoints" has been expanded to also include the concept of sub-meters -- i.e. as a means of grouping a number of different "meters" or "devices" together.

So, this is why we're thinking of re-naming "meteringPoints" -- in fact, now that I think about it more, "virtualDevices" may be a better name, as this encompass the idea of it being a billing point (i.e. something with an MPAN, which can be stored in the metadata object) or  of it being a collection of sub-devices.

Does that make sense?

(Updating issue description to "deviceGroups" with "virualDevices".)

**Comment by revenco-hk** [comment]
I agree that "Meter" is too narrow and that "Device" is better.

With regards to "MeterPoint" I think the notion of a point of metering/measurement is somewhat logically different from the grouping. Also there is already an "Entities" object which groups "MeterPoints" and "Meters" in the current specification which might suffice?

I think you will find that people will often want to group objects but for different reasons (household, project, sub-meters etc.) and therefore I think the base "MeterPoint" to "Meter/Device" relationship is a static building block at a lower level while groupings will be more adhoc and dynamic. Just my 2C :)

**Comment by andrewatamee** [comment]
Indeed, that makes sense -- perhaps we could use a type field in the grouping system to differentiate -- or, as you say, perhaps entities are better for this?

Needs more discussion, for sure -- perhaps again, we should move to https://groups.google.com/forum/#!forum/amon_data_format ?
